redis/src/cluster_async/mod.rs

```diff
@@ -1372,13 +1401,22 @@ where
         res
     }
 
+    /// Update relevant slots in the slots map based on the moved_redirect address
+    pub(crate) async fn update_slots_for_redirect_change(
+        _inner: Arc<InnerCore<C>>,
+        _moved_redirect: Option<RedirectNode>,
+    ) {
+        // TODO: Add implementation
+    }
+
     pub(crate) async fn check_topology_and_refresh_if_diff(
         inner: Arc<InnerCore<C>>,
         policy: &RefreshPolicy,
     ) -> bool {
         let topology_changed = Self::check_for_topology_diff(inner.clone()).await;
         if topology_changed {
-            let _ = Self::refresh_slots_and_subscriptions_with_retries(inner.clone(), policy).await;
+            let _ = Self::refresh_slots_and_subscriptions_with_retries(inner.clone(), policy, None)
```

Comment:
Can this function fail? why are we swallowing the return code?

Reply:
see https://github.com/amazon-contributing/redis-rs/pull/178/commits/21f8eb714bb04de41abb5d97dcbaead5710eba56